onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -124,9 +127,131 @@ def _parse_line(line):
+    return batch_data
+
+# Using augmented outputs to generate inputs to quantize.py
+def get_intermediate_outputs(augmented_model_path, inputs, average_mode='naive'):
+    '''
+    Gather intermediate model outputs after running inference
+        parameter model: path to augmented FP32 ONNX model
+        parameter inputs: list of loaded test inputs (or image matrices)
+        parameter average_mode: type 'naive' gives (ReduceMin, ReduceMax) pairs
+                                for each augmented node across test data sets, where
+                                the first element is a minimum of all ReduceMin values
+                                and the second element is a maximum of all ReduceMax
+                                values; the type 'smooth' yields a smooth average
+                                with anomalous values removed
+        return: dictionary mapping added node names to (ReduceMin, ReduceMax) pairs
+    '''
+    num_inputs = len(inputs)
+
+    # Conducting inference
+    session = onnxruntime.InferenceSession(augmented_model_path, None)
+    input_name = session.get_inputs()[0].name
+    intermediate_outputs = [session.run([], {input_name: inputs[i]}) for i in range(num_inputs)]
+
+    # Creating dictionary with output results from multiple test inputs
+    output_node_names = [session.get_outputs()[i].name for i in range(len(intermediate_outputs[0]))]
+    output_dicts = [dict(zip(output_node_names, intermediate_outputs[i])) for i in range(num_inputs)]
+    merged_dict = {}
+    for d in output_dicts:
+        for k, v in d.items():
+            merged_dict.setdefault(k, []).append(v)
+    added_output_node_names = output_node_names[1:]
+    node_names = [added_output_node_names[i].rpartition('_')[0] + ':0' for i in range(0, len(added_output_node_names), 2)] # output names
+
+    # Characterizing distribution of a node's values across test data sets
+    clean_merged_dict = dict((i, merged_dict[i]) for i in merged_dict if i != list(merged_dict.keys())[0])
+    if average_mode == 'naive':
+        pairs = [tuple([float(min(clean_merged_dict[added_output_node_names[i]])),
+                float(max(clean_merged_dict[added_output_node_names[i+1]]))])
+                for i in range(0, len(added_output_node_names), 2)]
+        final_dict = dict(zip(node_names, pairs))
+    elif average_mode == 'smooth':
+        # TODO: insert smooth averaging code
+        final_dict = {}
+    return final_dict
+
 def main():
-    # calibrate_loop()
-    pass
+    # Generating augmented ONNX model
+    model_path = input('Full filepath to model: ')
+    model = onnx.load(model_path)
+    augmented_model = augment_graph(model)
+    augmented_model_path = 'augmented_model.onnx'
+    onnx.save(augmented_model, augmented_model_path)
+    # Generating inputs for quantization
+    images_folder = input('Full filepath to images folder: ')
+    height = int(input('Image height (pixels): '))
```

Comment:
This seems a bit odd... can you not get it from the image?

Reply:
In [onnx_classify.py](https://github.com/ctuning/ck-mlperf/blob/master/program/image-classification-onnx-py/onnx_classify.py), the height and width were what the user wanted to resize all the images in the dataset to, since  the images could have different dimensions to begin with...any suggestions as for whether to leave these as command line args?

Reply:
Resolved – got image dimensions from session.